src/terminal/input/terminalInput.cpp

```diff
@@ -474,20 +475,15 @@ TerminalInput::OutputType TerminalInput::MakeOutput(const std::wstring_view& str
 // Return Value:
 // - Returns an empty optional if we didn't handle the key event and the caller can opt to handle it in some other way.
 // - Returns a string if we successfully translated it into a VT input sequence.
-TerminalInput::OutputType TerminalInput::HandleKey(const IInputEvent* const pInEvent)
+TerminalInput::OutputType TerminalInput::HandleKey(const INPUT_RECORD& event)
 {
-    if (!pInEvent)
-    {
-        return MakeUnhandled();
-    }
-
     // On key presses, prepare to translate to VT compatible sequences
-    if (pInEvent->EventType() != InputEventType::KeyEvent)
+    if (event.EventType != KEY_EVENT)
     {
         return MakeUnhandled();
     }
 
-    auto keyEvent = *static_cast<const KeyEvent* const>(pInEvent);
+    auto keyEvent = event.Event.KeyEvent;

```

Comment:
hmm would we benefit from an `auto&`?

Reply:
We modify the `keyEvent` unfortunately to decoalesce the repeated key events.